mu/logic.py

```diff
@@ -112,12 +111,42 @@ def get_settings_path():
     if not os.path.exists(settings_dir):
         settings_dir = os.path.join(DATA_DIR, settings_filename)
         if not os.path.exists(settings_dir):
-            with open(settings_dir, 'w') as f:
-                logger.debug('Creating settings file: {}'.format(settings_dir))
-                json.dump({}, f)
+            try:
+                with open(settings_dir, 'w') as f:
+                    logger.debug('Creating settings file: {}'.format(
+                                 settings_dir))
+                    json.dump({}, f)
+            except FileNotFoundError:
+                logger.error('Unable to create settings file: {}'.format(
+                             settings_dir))
     return settings_dir
 
 
+def get_workspace_dir():
```

Comment:
Briefly (I have dad duty things taking up my time this evening, will reply more fully tomorrow AM), yes: ensure you set up the test with a settings file somewhere temporary and Mock away get_settings_path to return a path to this new temp settings file. Then test to your hearts' content. Does this make sense..?